### DISCUSSION TITLE
Unity AndroidJavaException: java.lang.IllegalArgumentException: Couldn't find meta-data for provider with authority com.XXX.fileprovider<title>

### DISCUSSION THREAD

**Discussion by Andogeon** [community/community#72930] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

hello. I am sharing an Instagram feed with Unity, but when I call the FileProvider.getUriForFile() file function, an Error Unity at androidx.core.content.FileProvider.getFileProviderPathsMetaData(FileProvider.java:664) error appears. Is there any way to solve this??

Android Manifest.xml
`<?xml version="1.0" encoding="utf-8"?>
<manifest
    xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.unity3d.player"
    xmlns:tools="http://schemas.android.com/tools">

    <queries>
      <package android:name="com.kakao.talk" />
    </queries>

    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="com.google.android.gms.permission.AD_ID"
                     tools:node="remove"/>
  
    <application android:name="com.XXX.GlobalApplication">
        <activity android:name="com.unity3d.player.UnityPlayerActivity"
                  android:theme="@style/UnityThemeSelector"
                  android:exported="true">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>

           <intent-filter>
             <data android:host="kakaolink"
               android:scheme="kakao$XXXXXXXXXXXXXXXXXXXXXXXXXXX" />
           </intent-filter>
          
            <meta-data android:name="unityplayer.UnityActivity" android:value="true" />
            <meta-data android:name="google_analytics_adid_collection_enabled" android:value="false" />
        </activity>
            
          <provider
            android:name="androidx.core.content.FileProvider"
            android:authorities="com.XXX.provider"
            android:exported="false"
            android:grantUriPermissions="true">
            <meta-data
                android:name="android.support.FILE_PROVIDER_PATHS"
                android:resource="@xml/file_provider" />
          </provider>

      <receiver android:name="com.XXX.UnityShare"  android:exported="true">
        <intent-filter>
          <action android:name="android.intent.action.BOOT_COMPLETED"/>
          <action android:name="android.intent.action.INPUT_METHOD_CHANGED"/>
        </intent-filter>
      </receiver>
    </application>
</manifest>`

file_provider.xml
`<?xml version="1.0" encoding="utf-8"?>
<paths xmlns:android="http://schemas.android.com/apk/res/android">
  <external-path name='name' path='.'/>
</paths>``

Error Unity AndroidJavaException: java.lang.IllegalArgumentException: Couldn't find meta-data for provider with authority com.XXXX.fileprovider
Error Unity java.lang.IllegalArgumentException: Couldn't find meta-data for provider with authority com.XXX.fileprovider
Error Unity at androidx.core.content.FileProvider.getFileProviderPathsMetaData(FileProvider.java:664)

**Comment by mnicolas94** [comment]
@Andogeon Did you find a solution for this? I'm getting the same error.
